src/ray/object_manager/ownership_based_object_directory.cc

```diff
@@ -207,44 +190,29 @@ void OwnershipBasedObjectDirectory::SubscriptionCallback(
   it->second.subscribed = true;
 
   // Update entries for this object.
-  if (UpdateObjectLocations(reply, status, object_id, gcs_client_, mark_as_failed_,
-                            &it->second.current_object_locations, &it->second.spilled_url,
-                            &it->second.spilled_node_id, &it->second.object_size)) {
-    RAY_LOG(DEBUG) << "Pushing location updates to subscribers for object " << object_id
-                   << ": " << it->second.current_object_locations.size()
-                   << " locations, spilled_url: " << it->second.spilled_url
-                   << ", spilled node ID: " << it->second.spilled_node_id
-                   << ", object size: " << it->second.object_size;
-    metrics_num_object_location_updates_++;
-    // Copy the callbacks so that the callbacks can unsubscribe without interrupting
-    // looping over the callbacks.
-    auto callbacks = it->second.callbacks;
-    // Call all callbacks associated with the object id locations we have
-    // received.  This notifies the client even if the list of locations is
-    // empty, since this may indicate that the objects have been evicted from
-    // all nodes.
-    for (const auto &callback_pair : callbacks) {
-      // We can call the callback directly without worrying about invalidating caller
-      // iterators since this is already running in the subscription callback stack.
-      // See https://github.com/ray-project/ray/issues/2959.
-      callback_pair.second(object_id, it->second.current_object_locations,
-                           it->second.spilled_url, it->second.spilled_node_id,
-                           it->second.object_size);
-    }
-  }
-
-  // Only send the next long-polling RPC if the last one was successful.
-  // If the last RPC failed, we consider the object to have been freed.
-  if (status.ok()) {
-    auto worker_it = worker_rpc_clients_.find(worker_id);
-    rpc::GetObjectLocationsOwnerRequest request;
-    request.set_intended_worker_id(worker_id.Binary());
-    request.set_object_id(object_id.Binary());
-    request.set_last_version(reply.current_version());
-    worker_it->second->GetObjectLocationsOwner(
-        request,
-        std::bind(&OwnershipBasedObjectDirectory::SubscriptionCallback, this, object_id,
-                  worker_id, std::placeholders::_1, std::placeholders::_2));
+  UpdateObjectLocations(object_location_message, object_id, gcs_client_,
```

Comment:
subscription callback doesn't need to pass the status to UpdateObjectLocations anymore because it is only called when the publisher publishes a normal message now.